fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/service/RecalculateInterestPoster.java

```diff
@@ -22,27 +22,26 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.concurrent.Callable;
-import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
 import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.infrastructure.core.domain.FineractContext;
+import org.apache.fineract.infrastructure.core.service.ThreadLocalContextUtil;
 import org.apache.fineract.infrastructure.jobs.exception.JobExecutionException;
 
-@NoArgsConstructor
+@RequiredArgsConstructor
 @Slf4j
 public class RecalculateInterestPoster implements Callable<Void> {
 
+    @Setter
     private Collection<Long> loanIds;
-    private LoanWritePlatformService loanWritePlatformService;
-
-    public void setLoanIds(final Collection<Long> loanIds) {
-        this.loanIds = loanIds;
-    }
-
-    public void setLoanWritePlatformService(final LoanWritePlatformService loanWritePlatformService) {
-        this.loanWritePlatformService = loanWritePlatformService;
-    }
+    @Setter
+    private FineractContext fineractContext;
+    private final LoanWritePlatformService loanWritePlatformService;
 
     @Override
     public Void call() throws JobExecutionException {
+        ThreadLocalContextUtil.init(fineractContext);
```

Comment:
This is necessary to work correctly. new/ol threads may have no or wrong context.